MQ2AutoSize.cpp

```diff
@@ -1,627 +1,1387 @@
-// Usage:
-//	This plugin will automatically resize configured spawns to the specified
-//	size. You can configure it to only resize within a specific range and then
-//	resize back to normal when your distance moves out of that range.
-//	Current default range is set to 50 and may be changed via INI or cmd line
-//	NOTE:  These effects are CLIENT SIDE ONLY!
-//
-// Commands:
-//  /autosize				  - Toggles zone-wide  AutoSize on/off
-//  /autosize dist			- Toggles distance-based AutoSize on/off
-//  /autosize pc			  - Toggles AutoSize PC spawn types
-//  /autosize npc			 - Toggles AutoSize NPC spawn types
-//  /autosize pets			- Toggles AutoSize pet spawn types
-//  /autosize mercs		  - Toggles AutoSize mercenary spawn types
-//  /autosize mounts		 - Toggles AutoSize mounted player spawn types
-//  /autosize corpse		 - Toggles AutoSize corpse spawn types
-//  /autosize target		 - Resizes your target to sizetarget size
-//  /autosize everything	- Toggles AutoSize all spawn types
-//  /autosize self			- Toggles AutoSize for your character
-//  /autosize range #		- Sets range for distance-based AutoSize
-//
-//  (Valid sizes 1 to 250)
-//  /autosize size #		 - Sets default size for "everything"
-//  /autosize sizepc #	  - Sets size for PC spawn types
-//  /autosize sizenpc #	 - Sets size for NPC spawn types
-//  /autosize sizepets #	- Sets size for pet spawn types
-//  /autosize sizetarget # - Sets size for target parameter
-//  /autosize sizemercs #  - Sets size for mercenary spawn types
-//  /autosize sizemounts # - Sets size for mounted player spawn types
-//  /autosize sizecorpse # - Sets size for corpse spawn types
-//  /autosize sizeself #	- Sets size for your character
-//
-//  /autosize status		 - Display current plugin settings to chatwnd
-//  /autosize help			- Display command syntax to chatwnd
-//  /autosize save			- Save settings to INI file (auto on plugin unload)
-//  /autosize load			- Load settings from INI file (auto on plugin load)
-//  /autosize autosave	  - Automatically save settings to INI file when an option is toggled or size is set
-//
-//////////////////////////////////////////////////////////////////////////////
+// MQ2AutoSize.cpp : Resize spawns by distance (client only)
 
 #include <mq/Plugin.h>
 
+// Constants
 const char* MODULE_NAME = "MQ2AutoSize";
+const int SKIP_PULSES = 5;               // Controls the number of pulses to perform a radius-based resize
+const int MIN_SIZE = 1;                  // Minimum size value
+const int MAX_SIZE = 250;                // Maximum size value
+int FAR_CLIP_PLANE = 1000;               // Placeholder for EQ Far Clip Plane value
+const float OTHER_SIZE = 1.0f;           // Default size for other entities
+const float ZERO_SIZE = 0.0f;            // Size representing zero
+const int DEFAULT_COMMS = 0;             // Default comms value
+const int DEFAULT_OPT_ZONEWIDE = 2;      // Default option for zonewide
```

Comment:
lets make these constexpr, and should FAR_CLIP_PLANE be a constexpr as well?

Reply:
I've applied this and included FAR_CLIP_PLANE.